spec/kolekti_analizo/collector_spec.rb

```diff
@@ -57,10 +62,14 @@
     end
 
     describe 'default_value_from' do
-      let(:metric_list) { FactoryGirl.build(:analizo_metric_collector_list).raw }
+      let!(:metric_list) { FactoryGirl.build(:analizo_metric_collector_list).raw }
```

Comment:
On PHPMD collector we do mock a protected method. Look [here](https://github.com/mezuro/kolekti_cc_phpmd/blob/master/spec/kolekti/cc_phpmd/collector_spec.rb#L51). Sadly, when testing the `collect_metrics` method we do not mock it, the same problem. So, other collectors have the same problem.

I believe the reason we did not create a YAML file for Analizo is because it has a command that list the supported metrics and it's not a static file (we can get new metrics for free if there are any on a package update). And it seems a waste to generate a YAML file from the command output so we can use `Kolekti#parse_supported_metrics`.

I am not suggesting a refactoring right now, but asking to improve our test implementation. What we could do is to leave the way we're testing for the initialization test and on other tests that use those methods indirectly (like this one) we try the same approach as we did on PHPMD.

Since this is something not added by this PR and the tests are meaningful, we can accept the PR and create issues for this problem.

What do you think?